src/cephadm/cephadm.py

```diff
@@ -2580,18 +2600,23 @@ def get_container_mounts(ctx, fsid, daemon_type, daemon_id,
         mounts.update(CephIscsi.get_container_mounts(data_dir, log_dir))
 
     if daemon_type == Keepalived.daemon_type:
-        assert daemon_id
-        ident = DaemonIdentity(fsid, daemon_type, daemon_id)
         data_dir = get_data_dir(ident, ctx.data_dir)
         mounts.update(Keepalived.get_container_mounts(data_dir))
 
     if daemon_type == CustomContainer.daemon_type:
-        assert daemon_id
-        cc = CustomContainer.init(ctx, fsid, daemon_id)
-        ident = DaemonIdentity(fsid, daemon_type, daemon_id)
+        cc = CustomContainer.init(ctx, fsid, ident.daemon_id)
         data_dir = get_data_dir(ident, ctx.data_dir)
         mounts.update(cc.get_container_mounts(data_dir))
 
+    _update_podman_mounts(ctx, mounts)
+    return mounts
+
+
+def _update_podman_mounts(ctx: CephadmContext, mounts: Dict[str, str]) -> None:
+    """Update the given mounts dict with mounts specific to podman.
+    Should be called *after* all other mounts are added in order to avoid
+    overriding customized mounts.
```

Comment:
I'm a bit confused by this. If we want to make sure these mounts don't overwrite any of the other ones, wouldn't we want to do this first? So if any of the other mounts were on the same location they would overwrite the ones from here when the dict entry gets updated? It seems like the only one being set in this function checks first it isn't already in the mounts anyway so functionally it's the same, but the comment confuses me.

Reply:
I will simplify this comment.

Reply:
done